syncer/syncer.go

```diff
@@ -373,9 +376,18 @@ func (s *Syncer) syncLayer(ctx context.Context, layerID types.LayerID) (*types.L
 		}
 	} else {
 		s.logger.WithContext(ctx).With().Info("polling layer content", layerID)
-		if layer, err = s.getLayerFromPeers(ctx, layerID); err != nil {
+		var highest types.LayerID
+		layer, highest, err = s.getLayerFromPeers(ctx, layerID)
+		if err != nil {
 			return nil, err
 		}
+		if highest.Before(layerID) {
```

Comment:
i think this is quite a bit more logic/code than necessary to deal with this issue, here and in layers.go.

what do you think about changing the code in `layerBlocksReqReceiver` to only return blocks when the requested layer is lower than its processed layer?

Reply:
good suggestion, i considered it before but decided that i don't want to error and switch back to non-synced in some cases. but maybe that was wrong